serve.go

```diff
@@ -120,12 +120,12 @@ func (vs *version) serveError(w http.ResponseWriter, key string, err error) {
 	w.WriteHeader(http.StatusInternalServerError)
 }
 
-func shuffle(vs []string) []string {
+func shuffle(vs []string, disappeared []string) []string {
```

Comment:
This will return an arbitrary number of deleted nodes in addition to the live ones, which will change the behavior of proxy subtly. I think we want to try just <replication factor> peers, where dead peers are only considered if there aren't enough live peers to fulfill the replication factor.